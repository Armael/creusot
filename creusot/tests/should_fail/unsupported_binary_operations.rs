fn bit_and(a: u32, b: u32) -> u32 {
    a & b
}

fn bit_or(a: u32, b: u32) -> u32 {
    a | b
}

fn bit_xor(a: u32, b: u32) -> u32 {
    a ^ b
}
