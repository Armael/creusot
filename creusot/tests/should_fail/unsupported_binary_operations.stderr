warning: function `bit_and` is never used
 --> unsupported_binary_operations.rs:1:4
  |
1 | fn bit_and(a: u32, b: u32) -> u32 {
  |    ^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: function `bit_or` is never used
 --> unsupported_binary_operations.rs:5:4
  |
5 | fn bit_or(a: u32, b: u32) -> u32 {
  |    ^^^^^^

warning: function `bit_xor` is never used
 --> unsupported_binary_operations.rs:9:4
  |
9 | fn bit_xor(a: u32, b: u32) -> u32 {
  |    ^^^^^^^

warning: load the `creusot_contract` crate to enable resolution of mutable borrows.

error[creusot]: bitwise operations are currently unsupported
 --> unsupported_binary_operations.rs:2:5
  |
2 |     a & b
  |     ^^^^^

error: aborting due to previous error; 4 warnings emitted

